Add a link to the rust library for interacting with the remote API

plule opened the pull request:
Add a link to https://docs.rs/dfhack-remote/latest/dfhack_remote/index.html, a library using the remote API for the rust programming language. Note: the github repo is https://github.com/plule/dfhack-remote if you prefer linking to it rather than docs.rs

lethosor commented:
If you can easily cross-link between the two places, I'm ok with either. I'm used to package registries linking to the source repositories (e.g. PyPI, NPM). The only GitHub link I can find at https://docs.rs/dfhack-remote/latest/dfhack_remote/index.html is in a dropdown, but if that's useful to people used to that site, that's fine with me. Alternatively, we could link to both the docs + repo. A lot of these libraries don't have hosted docs.

myk002 commented:
Also, could you add a note to the Docs subsection in the `docs/changelog.txt` file? (At the top of the file, under the "Future" header)

lethosor commented on docs/Remote.rst:
```diff
@@ -79,6 +79,7 @@ from other (non-C++) languages, including:
 - `dfhack-client-qt <https://github.com/cvuchener/dfhack-client-qt>`_ for C++ with Qt
 - `dfhack-client-python <https://github.com/McArcady/dfhack-client-python>`_ for Python (adapted from :forums:`"Blendwarf" <178089>`)
 - `dfhack-client-java <https://github.com/McArcady/dfhack-client-java>`_ for Java
+- `dfhack-remote <https://docs.rs/dfhack-remote/latest/dfhack_remote/index.html>`_ for Rust
```
I think you can get around the link name conflict just by using a double underscore at the end instead of a single underscore. You might need to do it for the other dfhack-remote link too.

plule replied:
It fixed it, thanks!

plule commented:
> If you can easily cross-link between the two places, I'm ok with either. I'm used to package registries linking to the source repositories (e.g. PyPI, NPM). The only GitHub link I can find at https://docs.rs/dfhack-remote/latest/dfhack_remote/index.html is in a dropdown, but if that's useful to people used to that site, that's fine with me. Alternatively, we could link to both the docs + repo. A lot of these libraries don't have hosted docs, which is why their links just go to a repo.

Ok. I was mostly wondering about consistency, let's keep the docs.rs link then. I've added a github badge on the doc page, to get proper crosslinking, it looks like it's common to have this.

> Also, could you add a note to the Docs subsection in the docs/changelog.txt file? (At the top of the file, under the "Future" header)

Sure, done!

> Btw, the "dfhack-remote" link name is taken by the JavaScript client a few lines above your new link. That's why the docs build is failing. You might need to choose a slightly different name. dfhack-remote-rust?

Fixed with the double underscore suggested by lethosor.

I *think* that the other build issue is spurious, but not sure how to rerun. The link to the generated documentation per PR is really great btw.

myk002 reviewed (approved):
Looks good! Thanks! I'm re-running the verification build, then I'll merge once it passes.

myk002 commented:
The test failure is on me and I have that fixed elsewhere. Thanks!

plule commented:
@myk002 thank you!
